### PR TITLE
`redis` update `scan` function 

### PR DESCRIPTION
**Pull request by mtuchi**
## Summary

Update scan to `scan` function to scan whole database based on the provided pattern 

Ref: #710 

## Notes

- [x] scan now iterate the whole database
- [x] removed `cursor` parameter from scan 
- [x] We support count as an argument in options object. Incase users want bigger "pages". This allows for some optimization.
- [x] Removed default for `type` option
- [x] Add support for json type and map it to the redis internal type

## Review Checklist

- [x] Does the PR do what it claims to do?
- [x] Are there any unit tests? Should there be?
- [x] Is there a changeset associated with this PR? Should there be? Note that
      dev only changes don't need a changeset.

### CODE REVIEW

**Comment by mtuchi** [comment]
Hiya @josephjclark is first PR on redis improvements, I would love to hear your feedback on unit test if we need to add more

**Review by josephjclark — CHANGES_REQUESTED** [review]
looks great @mtuchi, let's add one more test and there's a couple of tiny suggestions

File: packages/redis/src/Adaptor.js [hunk 1]
```diff
@@ -47,8 +47,7 @@ const disconnect = async state => {
  * Options provided to the scan function
  * @typedef {Object} ScanOptions
  * @public
- * @property {integer} [cursor=0] - A numeric value used to continue the iteration from where it left off. Initially, you start with 0.
- * @property {string} [type='hash'] - Limits the keys returned to those of a specified type (e.g., string, list, set, hash, etc.).
+ * @property {string} type - Limits the keys returned to those of a specified type (e.g., string, list, set, hash, etc.).
```
[the remark this answers is not in the document]
Comment (mtuchi) [reply packages/redis/src/Adaptor.js]: Done

File: packages/redis/test/Adaptor.test.js [hunk 2]
```diff
@@ -254,6 +254,19 @@ describe('hset', () => {
 });
 
 describe('scan', () => {
+  it('should return json keys', async () => {
```
[the remark this answers is not in the document]
Comment (mtuchi) [reply packages/redis/test/Adaptor.test.js:275]: I have added the test, but i don't how to assert the cursor state i each iteration

[the remark this answers is not in the document]
Comment (mtuchi) [reply packages/redis/test/Adaptor.test.js:275]: Let me try this

**Comment by mtuchi** [comment]
Hiya @josephjclark i have created a `release/redis` branch. Should i change the base for this branch and update the changeset in `release/redis` branch ?

File: packages/redis/test/Adaptor.test.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (josephjclark) [reply packages/redis/test/Adaptor.test.js:275]: Great!

**Review by josephjclark — APPROVED** [review]

**Pull request merged**